esmgrids/cice_grid.py

```diff
@@ -135,24 +180,76 @@ def write(self, grid_filename, mask_filename):
         ulat[:] = np.deg2rad(self.y_u)
         ulon[:] = np.deg2rad(self.x_u)
 
-        if self.clon_t is not None:
-            clon_t[:] = np.deg2rad(self.clon_t)
-            clat_t[:] = np.deg2rad(self.clat_t)
-            clon_u[:] = np.deg2rad(self.clon_u)
-            clat_u[:] = np.deg2rad(self.clat_u)
-
         # Convert from m to cm.
         htn[:] = self.dx_tn[:] * 100.0
         hte[:] = self.dy_te[:] * 100.0
 
         angle[:] = np.deg2rad(self.angle_u[:])
         angleT[:] = np.deg2rad(self.angle_t[:])
 
-        f.close()
+        nc.close()
+
+    def write_mask(self, nc):
+        """
+        Write out CICE mask/kmt to netcdf. nc is a Netcdf4 Dataset obj
+        """
+
+        nc.createDimension("nx", self.num_lon_points)
+        nc.createDimension("ny", self.num_lat_points)
+        mask = nc.createVariable("kmt", "f8", dimensions=("ny", "nx"), compression="zlib", complevel=1)
+
+        mask.grid_mapping = "crs"
+        mask.standard_name = "sea_binary_mask"
+
+        # CICE uses 0 as masked, whereas internally we use 1 as masked.
+        mask[:] = 1 - self.mask_t
+        nc.close()
+
+
+def cice_from_mom(ocean_hgrid, ocean_mask, grid_file="grid.nc", mask_file="kmt.nc"):
+
+    mom = MomGrid.fromfile(ocean_hgrid, mask_file=ocean_mask)
+
+    cice = CiceGrid.fromgrid(mom)
+
+    cice.grid_nc = create_nc(grid_file)
+
+    # Add versioning information
+    cice.grid_nc.inputfile = f"{ocean_hgrid}"
+    cice.grid_nc.inputfile_md5 = md5sum(ocean_hgrid)
+    cice.grid_nc.history_command = f"python make_CICE_grid.py {ocean_hgrid} {ocean_mask}"
+
+    # Add the typical crs (i.e. WGS84/EPSG4326 , but in radians).
+    crs = cice.grid_nc.createVariable("crs", "S1")
+    crs.grid_mapping_name = "tripolar_latitude_longitude"
+    crs.crs_wkt = 'GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["radians",1,AUTHORITY["EPSG","9122"]],AXIS["Latitude",NORTH],AXIS["Longitude",EAST],AUTHORITY["EPSG","4326"]]'
```

Comment:
This is tricky. Our data relies on having 2d arrays of lat/lon (i.e. geolocation arrays) to be used. I haven't found a way to describe it using a proj code or a grid_mapping description.

So the crs here applies to the geolocation arrays. Without the geolocation arrays it doesn't really make sense.

So its the _crs_wkt_ that matters here, thats the information that could (in theory) be used by some piece of analysis software. I just tried to give a descriptive _grid_mapping_ name.  

The assumptions are that the input grid is a tripolar input grid in WGS84. The whole thing falls over if that is not true [because of the wrapping and folding of the grid.](https://github.com/COSIMA/esmgrids/blob/a93cdbb12b1f6f520500c31b03fa3bfb882a3eab/esmgrids/mom_grid.py#L101) So one day we will need to extend this code for regional grids.

Reply:
>  haven't found a way to describe it using a proj code or a grid_mapping description.

So should the `grid_mapping_name` attribute be removed?

Reply:
ok done